kale/evaluate/uncertainty_metrics.py

```diff
@@ -0,0 +1,1036 @@
+        entry = entry[1:-1]
+        bounds.append([float(i) for i in entry.split(",")])
+    return bounds
+
+
+def evaluate_correlations(
+    bin_predictions: Dict[str, pd.DataFrame],
+    uncertainty_error_pairs: List[Tuple[str, str, str]],
+    cmaps: List[Dict[Any, Any]],
+    num_bins: int,
+    conf_invert_info: List[Tuple[str, bool]],
+    num_folds: int = 8,
+    pixel_to_mm_scale: float = 1,
+    combine_middle_bins: bool = False,
+    save_path: Optional[str] = None,
+    to_log: bool = False,
+) -> Dict[str, Dict[str, Dict[str, Any]]]:
+    """
+    Calculates the correlation between error and uncertainty for each bin and for each landmark,
+    using a piece-wise linear regression model.
+
+    Args:
+        bin_predictions: A dictionary of Pandas DataFrames containing model predictions for each testing fold.
+        uncertainty_error_pairs: A list of tuples specifying the names of the uncertainty,
+            error, and uncertainty inversion keys for each pair.
+        cmaps: A dictionary of colour maps to use for plotting the results.
+        num_bins: The number of quantile bins to divide the data into.
+        conf_invert_info: A list of tuples specifying whether to invert the uncertainty values for each method.
+        num_folds: The number of folds to use for cross-validation (default: 8).
+        pixel_to_mm_scale: The scale factor to convert from pixels to millimeters (default: 1).
+        combine_middle_bins: Whether to combine the middle bins into one bin (default: False).
+        save_path: The path to save the correlation plots (default: None).
+        to_log: Whether to use logarithmic scaling for the x and y axes of the plots (default: False).
+
+    Returns:
+        A dictionary containing the correlation statistics for each model and uncertainty method.
+        The dictionary has the following structure:
+        {
+            <model_name>: {
+                <uncertainty_name>: {
+                    "all_folds": {
+                        "r": <correlation coefficient>,
+                        "p": <p-value>,
+                        "fit_params": <regression line parameters>,
+                        "ci": <confidence intervals for the regression line parameters>
+                    },
+                    "quantiles": {
+                        <quantile_index>: {
+                            "r": <correlation coefficient>,
+                            "p": <p-value>,
+                            "fit_params": <regression line parameters>,
+                            "ci": <confidence intervals for the regression line parameters>
+                        }
+                    }
+                }
+            }
+        }
+        The "all_folds" key contains the correlation statistics for all testing folds combined.
+        The "quantiles" key contains the correlation statistics for each quantile bin separately.
+    """
+    from kale.interpret.uncertainty_quantiles import fit_line_with_ci
```

Comment:
Why here and not at the top? Consistent style with other APIs is needed.

Reply:
circular import issue

Reply:
in discussion: https://github.com/pykale/pykale/discussions/392